Added note on env vars (#31237)
vercel/next.js: docs/api-reference/next/server.md

The `redirect()` method uses a `307` by default, instead of a `302` temporary redirect, meaning your requests will _always_ be preserved as `POST` requests.

### Accessing environment variables

> `process.env` holds an object with all environment variables used in Middleware, for both production and development environments. If you are not using any environment variables in your Middleware, the `process.env` object will be empty. `process.env` will only include environment variables _actually_ used.

Reviewer: ```suggestion
`process.env` can be used to access [environment variables](docs/basic-features/environment-variables.md) from Middleware. However, these are evaluated at build time so only environment variables _actually_ used will be included.
```

Its not clear what production and development means here (Vercel project settings or .env files??)

I think we can simplify this since it pertains to middleware only.

> Any variables in `process.env` can be accessed directly, and **cannot** be destructured:

Reviewer: ```suggestion
Any variables in `process.env` must be accessed directly, and **cannot** be destructured:
```

```ts
// This will not work
const { NODE_ENV } = process.env

> console.log(NODE_ENV)

Reviewer 2: ```suggestion
// NODE_ENV is `undefined`
console.log(NODE_ENV)
```

> // Empty object

Reviewer 2: ```suggestion
// process.env is `{}`
```

console.log(process.env)

> // Accessed directly, and not destructured works

Reviewer 2: ```suggestion
// Accessed directly, and not destructured works. process.env.NODE_ENV is `"development"` or `"production"`
```

The `redirect()` method uses a `307` by default, instead of a `302` temporary redirect, meaning your requests will _always_ be preserved as `POST` requests.

> ### Accessing environment variables

Reviewer 3: Given the above heading is a question - should we do "How do I access Environment Variables?"

Reviewer 3: Moved to inline suggestion

> `process.env` can be used to access [Environment Variables](docs/basic-features/environment-variables.md) from Middleware. These are evaluated at build time, so only environment variables _actually_ used will be included.

Reviewer: ```suggestion
`process.env` can be used to access [Environment Variables](/docs/basic-features/environment-variables.md) from Middleware. These are evaluated at build time, so only environment variables _actually_ used will be included.
```

> `process.env` can be used to access [environment variables](docs/basic-features/environment-variables.md) from Middleware. However, these are evaluated at build time so only environment variables _actually_ used will be included.

Reviewer 3: ```suggestion
`process.env` can be used to access [Environment Variables](docs/basic-features/environment-variables.md) from Middleware. These are evaluated at build time, so only environment variables _actually_ used will be included.
```

Any variables in `process.env` must be accessed directly, and **cannot** be destructured:

```ts
// This will not work
const { NODE_ENV } = process.env
// NODE_ENV is `undefined`
console.log(NODE_ENV)
// process.env is `{}`
console.log(process.env)

> // Accessed directly, and not destructured works. process.env.NODE_ENV is `"development"` or `"production"`

Reviewer 3: Should we start by showing what works?

Reviewer 3: ```
// This works when accessed directly. process.env.NODE_ENV is `"development"` or `"production"`
```

Author: That's a good idea. I like that way of thinking 🙌

> ### Accessing environment variables

Reviewer 3: ```suggestion
### How do I access Environment Variables?
```
